### PR TITLE
[Python API] Improve configuration files

### PR DESCRIPTION
**Pull request by akuporos**
### Details:
 - `Python API Checks` job now checks code-style also for new Python API, ngraph Python API
 - merged tox.ini to setup.cfg
 - removed Jenkins files for ngraph-onnx validation as it isn't run there.
 - updated code-style in samples and setup.py according to flake8.
 - updated requirements_test.txt -- extended with content from [this file](https://github.com/openvinotoolkit/openvino/blob/master/src/bindings/python/src/compatibility/openvino/requirements_dev.txt), removed fixed versions.

### Tickets:
 - 77643

### CODE REVIEW

File: src/bindings/python/setup.cfg [hunk 1]
```diff
@@ -20,4 +56,9 @@ disable_error_code = attr-defined
...
 show_error_context = True
 show_absolute_path = True
-pretty = True
\ No newline at end of file
+pretty = True
+follow_imports=normal
+disallow_untyped_defs = True
+disallow_untyped_calls = True
+check_untyped_defs = True
+show_none_errors  = True
```
Comment (jiwaszki) [inline src/bindings/python/setup.cfg:64]: New line?

File: src/bindings/python/setup.cfg [hunk 2]
```diff
@@ -1,15 +1,51 @@
...
+  OpenVINO_DIR = {env:OpenVINO_DIR}
+passenv =
+  http_proxy
+  https_proxy
+commands=
+  {envbindir}/python setup.py bdist_wheel
+  {envbindir}/pip install --no-index --pre --find-links=dist/ openvino
+  # flake8 {posargs:src/ setup.py} --exclude=src/compatibility/openvino
+  # flake8 --ignore=D100,D101,D102,D103,D104,D105,D107,D212,W503 tests/ tests_compatibility/ --exclude=tests_compatibility/test_inference_engine # ignore lack of docs in tests
+  # mypy --config-file=tox.ini {posargs:src/} --exclude=src/compatibility/openvino
```
Comment (jiwaszki) [inline src/bindings/python/setup.cfg:22]: Why are flake and mypy muted?

File: src/bindings/python/setup.cfg [hunk 3]
```diff
@@ -1,15 +1,51 @@
...
+envlist = py3
+
+[testenv]
+skipdist=True
+skip_install=True
+deps =
+  -rrequirements.txt
+  -rrequirements_test.txt
+setenv =
+  NGRAPH_BACKEND = {env:NGRAPH_BACKEND:"CPU"}
```
Comment (jiwaszki) [inline src/bindings/python/setup.cfg:11]:
```suggestion
  OV_BACKEND = {env:OV_BACKEND:"CPU"}
```

File: src/bindings/python/setup.cfg [hunk 4]
```diff
@@ -1,15 +1,51 @@
...
+  http_proxy
+  https_proxy
+commands=
+  {envbindir}/python setup.py bdist_wheel
+  {envbindir}/pip install --no-index --pre --find-links=dist/ openvino
+  # flake8 {posargs:src/ setup.py} --exclude=src/compatibility/openvino
+  # flake8 --ignore=D100,D101,D102,D103,D104,D105,D107,D212,W503 tests/ tests_compatibility/ --exclude=tests_compatibility/test_inference_engine # ignore lack of docs in tests
+  # mypy --config-file=tox.ini {posargs:src/} --exclude=src/compatibility/openvino
+  pytest --backend={env:NGRAPH_BACKEND} tests -v -k 'not _cuda' --ignore=tests/test_onnx/test_zoo_models.py --ignore=tests/test_utils --ignore=tests/test_inference_engine
+  pytest --backend={env:NGRAPH_BACKEND} tests_compatibility/test_ngraph -v -k 'not _cuda' --ignore=tests_compatibility/test_onnx/test_zoo_models.py
```
Comment (jiwaszki) [inline src/bindings/python/setup.cfg:24]:
```suggestion
  pytest --backend={env:OV_BACKEND} tests -v -k 'not _cuda' --ignore=tests/test_onnx/test_zoo_models.py --ignore=tests/test_utils --ignore=tests/test_inference_engine
  pytest --backend={env:OV_BACKEND} tests_compatibility/test_ngraph -v -k 'not _cuda' --ignore=tests_compatibility/test_onnx/test_zoo_models.py
```
Is ignoring `tests/test_inference_engine` planned?

File: src/bindings/python/setup.cfg [hunk 5]
```diff
@@ -1,15 +1,51 @@
...
+
+[testenv:zoo_models]
+commands=
+  {envbindir}/python setup.py bdist_wheel
+  {envbindir}/pip install --no-index --pre --find-links=dist/ openvino
+  pytest --backend={env:NGRAPH_BACKEND} tests/test_onnx/test_zoo_models.py -v -n 4 --forked -k 'not _cuda' --model_zoo_xfail
```
Comment (jiwaszki) [inline src/bindings/python/setup.cfg:30]:
```suggestion
  pytest --backend={env:OV_BACKEND} tests/test_onnx/test_zoo_models.py -v -n 4 --forked -k 'not _cuda' --model_zoo_xfail
```

File: src/bindings/python/setup.cfg (hunk 1 above)
Comment (akuporos) [reply src/bindings/python/setup.cfg]: done

File: src/bindings/python/setup.cfg (hunk 2 above)
Comment (akuporos) [reply src/bindings/python/setup.cfg]: I'm conducting some experiments :)

**Review by azhogov — APPROVED** [review]

**Review by jiwaszki — APPROVED** [review]
Can't wait to see it up and running! 🥇

**Review by AlexeyLebedev1 — APPROVED** [review]

File: .github/workflows/py_checks.yml [hunk 6]
```diff
@@ -66,10 +99,24 @@ jobs:
...
         run: python -m mypy ./ --config-file ./setup.cfg
         working-directory: src/bindings/python/src/compatibility/openvino
+      # nGraph Python API mypy check
+      - name: Run MyPy nGraph Python API
+        run: python -m mypy ./src/compatibility/ngraph --config-file ./setup.cfg
+        working-directory: src/bindings/python
+      # Python API 2.0 mypy check
+      - name: Run MyPy on Python API 2.0
+        run: python -m mypy ./src/openvino --config-file ./setup.cfg
+        working-directory: src/bindings/python
```
[the remark this answers is not in the document]
Comment (akuporos) [reply .github/workflows/py_checks.yml]: thanx, done

**Pull request merged**